### PR TITLE
Start skipping helix ready tests on azdo runs

### PR DESCRIPTION
**Pull request by HaoK**
Skips all tests on azdo by default, selectively enables test projects that currently don't work on helix (`BuildHelixPayload = false`).  

Note: we only run ubuntu and windows helix queues on every PR, so we would no longer be running mac OS tests as frequently

Per discussion in build sync: initial plan will be:
Required for every PR: Windows 10 and Ubuntu
Rolling builds for all other helix queues

### CODE REVIEW

**Comment by HaoK** [comment]
Test count with this change goes to: 25889 total reported on azdo compared to 115806 today

File: src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
     <!-- Tests do not work on Helix or when bin/ directory is not in project directory due to undeclared dependency on test content. -->
     <BuildHelixPayload>false</BuildHelixPayload>
+    <SkipTests>false</SkipTests>
```
Comment (Tratcher) [inline src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: Can't you just query BuildHelixPayload?

**Comment by Tratcher** [comment]
Are all of the tests that are now skipped on azdo part of a blocking helix run, or are some of them still only run in the optional helix run? They should be required on helix before being disabled on azdo, not just optional.

File: src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj (hunk 1 above)
Comment (HaoK) [reply src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: Possibly, what are you suggesting?

**Comment by HaoK** [comment]
The ubuntu queue is a blocking PR check, I believe there are some windows only tests that are being run in the optional helix queue still, we probably should make all (or some) windows queues required as part of this switch

File: src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj (hunk 1 above)
Comment (Tratcher) [reply src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: Is it possible to define skip tests globally based on BuildHelixPayload?

Comment (HaoK) [reply src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: I'm not a msbuild pro at all, my naive understanding is that its just evaluating files in order, so there's no way to default it dynamically globally so when its turned off later in a particular project it just works, which is why I just went with blasting it both flags to false.  There's no way to defer evaluation by reference like that, I don't think.

**Comment by HaoK** [comment]
Adding Win10 and OSX1014 to the mandatory helix green checks to match what we had in azdo

**Comment by HaoK** [comment]
@aspnet/build everything seems to be passing now, I think this PR is ready for final review

File: src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj (hunk 1 above)
Comment (dougbu) [reply src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: @HaoK suggest you move the setting to the root Directory.Build.targets file (after line 110 or so, where `$(BuildHelixPayload)` gets its default) and change the condition to
``` msbuild
<SkipTests Condition="'$(SkipHelixReadyTests)' == 'true' AND '$(BuildHelixPayload)' == 'true'">true</SkipTests>
```

**Comment by dougbu** [comment]
BTW this looks fine to me but I'd like someone more familiar w/ Helix to review more closely

File: src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj (hunk 1 above)
Comment (HaoK) [reply src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: awesome this is much better!

**Comment by dougbu** [comment]
FYI `CannotInvokeJSInvokableMethodsWithWrongNumberOfArguments` has had a flaky history according to https://github.com/search?q=CannotInvokeJSInvokableMethodsWithWrongNumberOfArguments&type=Issues but all four related issues are closed.  This looks like something new&hellip;

**Comment by dougbu** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=446311

**Comment by HaoK** [comment]
Now that https://github.com/dotnet/aspnetcore/pull/18274 has been merged, logging should be available on helix runs for test bugging, we should be ready for final sign on off on this PR @dotnet/aspnet-build

**Review by dougbu — APPROVED** [review]
Looks good and I suspect the macOS failure is a temporary network issue. Retried that job&hellip;

**Pull request merged**